### PR TITLE
Fix linking and usage of htslib, htslib 1.13 -> 1.16

### PR DESCRIPTION
**Pull request by fredlas**
It's not clear how, but htslib - as configured and installed by this repo until now - had started segfaulting upon simply reading the header of a newly opened file. Code that was previously working is doing that, despite the htslib version not having changed. All I can guess is that some Debian package that htslib in turn depends on got upgraded, and introduced this weirdness.

This PR switches to a newer version of htslib (1.16), links to it statically, and also overall simplifies the Makefile, for less places for weird behavior like this to sneak in. I also yanked most of the dependency setup stuff out of the Dockerfile into a shell script that the Dockerfile now calls, to make getting started with manual local development easier.

### CODE REVIEW

**Comment by fredlas** [comment]
Looks like the Dockerfile moved since I last looked, and it is now duplicated in two places. So, I just wrote my new version into both locations - someone who better knows the intentions for this repo's Docker usage can pick the right one to keep.

File: fastqpreprocessing/src/htslib_tagsort.cpp [hunk 1]
```diff
@@ -316,7 +316,9 @@ class AlignmentReader
     unsigned int cur_num_read = 0;
     while (cur_num_read < options_.alignments_per_batch)
     {
-      if (sam_read1(sam_file_ptr_, bam_hdr_, aln_arr_[thread_index][cur_num_read]) == 0)
+      // The documentation claims 0 means success, but I see code using >= and even >,
+      // so it looks like actually positive means success. I guess >= is safest, then.
+      if (sam_read1(sam_file_ptr_, bam_hdr_, aln_arr_[thread_index][cur_num_read]) >= 0)
```
Comment (fredlas) [inline fastqpreprocessing/src/htslib_tagsort.cpp:321]: (I ended up splitting this out into a separate PR)

File: fastqpreprocessing/src/htslib_tagsort.cpp [hunk 2]
```diff
@@ -462,3 +464,9 @@ std::vector<std::string> create_sorted_file_splits_htslib(INPUT_OPTIONS_TAGSORT
   return alignment_reader.partial_filenames();
 }  // function
 
+void crash(std::string msg)
```
Comment (fredlas) [inline fastqpreprocessing/src/htslib_tagsort.cpp:467]: In cleaning up the build process, I realized TagSort can be completely detached from the fastq programs, with crash() from utilities.cpp being the sole exception. With crash() duplicated into here, TagSort could be moved into its own separate directory.

**Review by jessicaway — APPROVED** [review]
Looks good! Thank you @fredlas!

**Comment by fredlas** [comment]
Great, thanks!

**Pull request merged**